File: problem1.js
function padding(S,target) {
    const pad = ' '.repeat(target);
    return pad + S;
}
var p = prompt("Enter the number of rows: ");
var r = parseInt(p);
for (var i=0;i<r;i++) {
    var stars = '*'.repeat(2 * i + 1);
    console.log(padding(stars, r-i));
}
for (var i=r-2; i>=0; i--) {
    let stars = '*'.repeat(2 * i + 1);
    console.log(padding(stars, r-i));
}

File: problem1.py
def padding(S,target):
     pad = ' ' * (target)
     return pad + S
r=int(input("enter number of rows"))
i=0
while i<r:
    palin='*' * ((2*i)+1)
    print(padding(palin,r-i))
    i+=1
i=r-2
while i>=0:
    palin='*' * ((2*i)+1)
    print(padding(palin,r-i))
    i-=1